### ISSUE TITLE
Bad training

### ISSUE THREAD

**Issue by LionelLeee** [AlexeyAB/darknet#5645]
Follow the author's steps to train your own data set (3 types) with yolov4. But the map is 0 after 10000 iterations. Loss has not changed.
![image](https://user-images.githubusercontent.com/38831106/82135625-57177600-9837-11ea-9991-dcbf95c8c781.png)
why?

**Comment by AlexeyAB** [comment]
Wrong dataset? Run training with `-show_imgs` flag, do you see correct bboxes?

**Comment by LionelLeee** [comment]
The dataset is composed of people, car, bus, truck, bicycle, motorcycle in the coco dataset.Run training with -show_imgs flag, I can get correct bboxes.

**Comment by AlexeyAB** [comment]
* What command do you use for training?
* Set `valid=train.txt` in `obj.data` file and run `./darknet detector map ...` what mAP do you get?

* show such screenshot 
![image](https://user-images.githubusercontent.com/4096485/82151535-97411d80-9864-11ea-8f25-a536c5845596.png)

* Attach your cfg file in zip

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82151737-9b366500-988f-11ea-9963-5a1f6640b387.png)
this is my screenshot.
My cfg file is as follows：
[cpmmodel.zip](https://github.com/AlexeyAB/darknet/files/4640417/cpmmodel.zip)

**Comment by AlexeyAB** [comment]
* Show several examples of txt-label files.

* Show content of files bad.list and bad_label.list

> Run training with -show_imgs flag, I can get correct bboxes.

* Show screenshot

* Show training log with several avg loss and other lines

**Comment by LionelLeee** [comment]
txt-label files：
![image](https://user-images.githubusercontent.com/38831106/82152179-20227e00-9892-11ea-9b41-3f1d1f5f92e6.png)
![image](https://user-images.githubusercontent.com/38831106/82152187-2dd80380-9892-11ea-9e00-1b82f67d5646.png)
![image](https://user-images.githubusercontent.com/38831106/82152196-392b2f00-9892-11ea-8435-fe7c72e13e20.png)
Did not generate files bad.list and bad_label.list
the follows is screenshot of  training with -show_imgs flag
![image](https://user-images.githubusercontent.com/38831106/82152297-c078a280-9892-11ea-8e2f-efc3bd8a603a.png)
 the follows is training log with several avg loss,but after running for a period of time, it will change the taste class to 0, and the GIOU is also 0
![image](https://user-images.githubusercontent.com/38831106/82152262-90c99a80-9892-11ea-902a-480407bbab2c.png)
@AlexeyAB

**Comment by AlexeyAB** [comment]
* Set `valid=train.txt` in `obj.data` file and run `./darknet detector map ...` what mAP do you get?

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82152710-c58b2100-9895-11ea-9ae0-c4696cf00104.png)
Is it similar to the picture above?It's running, it's slow

**Comment by uzairrizwan** [comment]
@AlexeyAB I am having the same issue. I have verified my dataset, .cfg file, .data file, .names file and all other required files, several times. But no success. The MaP is always 0, and the model doesn't give any predictions/detections.

**Comment by aicukltd** [comment]
I was having a very similar problem until I used the CFG file from the YOLO_MARK repository.

https://github.com/AlexeyAB/Yolo_mark/blob/master/x64/Release/yolo-obj.cfg

I followed the instructions in the YOLO_MARK repository (replacing classes with my count and filters with (classes + 5)*5 and I am getting mAP of 88% and AVG LOSS of 0.6~

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82225994-81148980-9958-11ea-88c5-d47faa24691b.png)
I used darknet.exe detector map for one day but it was extremely slow, it increased by about 4 every 5 minutes.
@AlexeyAB

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82226388-07c96680-9959-11ea-9df9-bdef22c2146e.png)
Is it normal that the above picture appears after a period of operation?

**Comment by aicukltd** [comment]
@LionelLeee try my instructions and let me know if you get any training results?

My output looked the same as your's until I used the other .cfg

**Comment by AlexeyAB** [comment]
@LionelLeee 

> ![image](https://user-images.githubusercontent.com/38831106/82152710-c58b2100-9895-11ea-9ae0-c4696cf00104.png)
> Is it similar to the picture above?It's running, it's slow

You should use `/backup/cpmodel_last.weights` instead of `yolov4.conv.137` weights for mAP calculation. https://github.com/AlexeyAB/darknet#when-should-i-stop-training

**Comment by LionelLeee** [comment]
@AlexeyAB 
Thanks, do you have any solutions to my problem?

**Comment by AlexeyAB** [comment]
Yes, do everything according to the manual and do not make mistakes.
So what is the mAP value?

**Comment by aicukltd** [comment]
On this note then @AlexeyAB can you explain why I got the exact same output as @LionelLeee using the yolo-v4.cfg from the latest commit (unchanged other than classes and filters)? Then when I switched to the CFG from the YOLO MARK repo everything is working perfectly?

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82228744-1bc29780-995c-11ea-9dc1-7d916b290b87.png)
this is map value.
@AlexeyAB

**Comment by AlexeyAB** [comment]
@aicukltd @LionelLeee Try to train from the begining by using 1 GPU, do you get mAP higher than 0?

**Comment by LionelLeee** [comment]
I used a gpu to get the above map.
@AlexeyAB

**Comment by AlexeyAB** [comment]
And show mAP that you get by usig this command
`./darknet detector map F:/MSCOCO/coco_f.data yolov4-custom.cfg backup/yolov4-custom_last.weights -iou_thresh 0.01`

**Comment by LionelLeee** [comment]
![image](https://user-images.githubusercontent.com/38831106/82233918-45cb8800-9963-11ea-8e3f-562b6c4b687b.png)

**Comment by AlexeyAB** [comment]
I just trained default MSCOCO-2014 dataset for 10 minutes: https://github.com/AlexeyAB/darknet/blob/master/scripts/get_coco_dataset.sh

By using this command
`darknet.exe detector train F:/MSCOCO/coco_f.data yolov4-custom.cfg yolov4.conv.137 -map `

With `width=416 height=416 batch=2 subdivisions=1 max_batches=2000` just for `2000` iterations for 10 minutes with final loss=~24.0 

* cfg: [yolov4-custom.cfg.txt](https://github.com/AlexeyAB/darknet/files/4645284/yolov4-custom.cfg.txt)

* pre-trained weights: https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v3_optimal/yolov4.conv.137

With such content in the `F:/MSCOCO/coco_f.data` file
```
classes= 80
train  = F:/MSCOCO/trainvalno5k_f.txt
valid = F:/MSCOCO/5k_f.txt
#valid = F:/MSCOCO/val2017_f.list
#valid = F:/MSCOCO/testdev2017_f.txt
names = data/coco.names
backup = backup
eval=coco
```

And get **non-zero** mAP@0.01 = `0.17%`
by using this command:
`darknet.exe detector map F:/MSCOCO/coco_f.data yolov4-custom.cfg backup/yolov4-custom_last.weights -iou_thresh 0.01`

![image](https://user-images.githubusercontent.com/4096485/82235279-2e78a480-993b-11ea-8108-d061268546e9.png)

![image](https://user-images.githubusercontent.com/4096485/82235314-36384900-993b-11ea-8e00-d3ded579e5c0.png)

----

I dont know how do you get `mAP@0.01 = 0` after 10 000 iterations with batch=64.

**Comment by LionelLeee** [comment]
One very strange thing, I changed to the cfg file of yolov3-tiny, with the same data set, it is working properly, and can get the correct map. I used the original cfg, using different data sets, he can also work normally, can also get a normal map. But as long as it is the original cfg and the original data set, he cannot get a normal map.It will also become the following after running for a period of time, but this does not happen in the above two training processes.
![chart_cpmtest](https://user-images.githubusercontent.com/38831106/82393326-9da7e300-9a78-11ea-96e9-4f8c207ac9fb.png)
![chart_kitti](https://user-images.githubusercontent.com/38831106/82393438-efe90400-9a78-11ea-97f1-a68e567f3cd6.png)
![image](https://user-images.githubusercontent.com/38831106/82393454-faa39900-9a78-11ea-8361-2eb7d34a4833.png)
I am very confused, you can explain why this is？
@AlexeyAB

**Comment by AlexeyAB** [comment]
What is is the original cfg and the original data set?

**Comment by LionelLeee** [comment]
It is the cpmmodel.cfg that was sent to you above, it comes from the modification of yolov4-custom, the data set is to select 6 categories from the coco dataset（people, car, bus, truck, bicycle, motorcycle）
